source/hp/dof_handler.cc

```diff
@@ -1349,7 +1349,8 @@ namespace hp
     // use the cell-wise access.
     active_cell_iterator cell = begin_active(), endc = end();
     for (unsigned int i = 0; cell != endc; ++cell, ++i)
-      active_fe_indices[i] = cell->active_fe_index();
+      if (cell->is_locally_owned())
+        active_fe_indices[i] = cell->active_fe_index();
```

Comment:
@masterleinad has tried very hard to convert all of these loops into range-based for loops. Can you do the same here?